pkg/download/automation/automation.go

```diff
@@ -101,6 +105,41 @@ func (d *Downloader) Download(projectName string, automationTypes ...config.Auto
 type NoopAutomationDownloader struct {
 }
 
+func createTemplateFromRawJSON(obj automationClient.Response) (t template.Template, extractedName string) {
+	configId := obj.Id
+
+	var data map[string]interface{}
+	err := json.Unmarshal(obj.Data, &data)
+	if err != nil {
+		log.Warn("Failed to sanitize downloaded JSON for config %v - template may need manual cleanup: %v", configId, err)
+		return template.NewDownloadTemplate(configId, configId, string(obj.Data)), configId
+	}
+
+	// remove properties not necessary for upload
+	delete(data, "id")
+	delete(data, "modificationInfo")
+	delete(data, "lastExecution")
+
+	// extract 'title' as name
+	configName := configId
+	if title, ok := data["title"]; ok {
+		configName = fmt.Sprintf("%v", title)
+		data["title"] = "{{.name}}"
+	}
+
+	var content []byte
+	if modifiedJson, err := json.Marshal(data); err == nil {
+		content = modifiedJson
+	} else {
+		log.Warn("Failed to sanitize downloaded JSON for config %v - template may need manual cleanup: %v", configId, err)
+		content = obj.Data
+	}
+	content, _ = jsonutils.MarshalIndent(content)
```

Comment:
Why do we even return the error if we don't use it? It's only used in tests :/

Reply:
Possibly removing the error return for MarshalIndent (or reacting to it in places where we're not sure marshalling will work) to be done in unrelated followup PR